roles/publish-charm/tasks/main.yaml

```diff
@@ -0,0 +1,19 @@
+- name: set channel
+  when: needs_charm_build and charm_build_output.rc is defined and charm_build_output.rc == 0 and build_type == "charmcraft"
+  register: channel
+  args:
+    executable: /bin/bash
+  shell: |
+    set -x
+    branch={{ zuul.branch }}
+    if [[ $branch == "main" ]]; then echo "latest/edge"; else echo ${branch#stable/}/edge; fi
```

Comment:
You'll probably want to publish to something other than edge over time.  e.g. stable branches, etc.

Reply:
New variable publish_channel can be introduced which holds the channel information. This variable can stay in respective projects in osci.yaml (OR) a map of charm name, publish_channel, charmcraft_channel can exist in this project )similar to [1]). 
I prefer the first approach because a change in osci.yaml will trigger CI builds and ultimately promote pipeline to publish.
Your thoughts?

[1] https://github.com/openstack-charmers/release-tools/blob/master/lp-builder-config/openstack.yaml

Reply:
Done